db/commitlog/commitlog.cc

```diff
@@ -2658,6 +2675,10 @@ future<> db::commitlog::force_new_active_segment() noexcept {
     co_await _segment_manager->force_new_active_segment();
 }
 
+future<> db::commitlog::wait_for_pending_deletes() noexcept {
+    co_await _segment_manager->wait_for_pending_deletes();
```

Comment:
It's a bit wasteful for this function to be a coroutine.
What's wrong with `return _segment_manager->wait_for_pending_deletes();`?
Yet it rarely called so it's not a big deal.